### PR TITLE
compatibility with vite

### PR DESCRIPTION
**Pull request by wikigods**
Support for vite
Today looking for a solution to use vite in the modules I found a temporary solution, I am open to suggestions for a better implementation in code. (Sorry my english is not very good) 
attached images of the result I obtained
Screenshot_118
Screenshot_119
Screenshot_120
Screenshot_121

### CODE REVIEW

**Comment by kylecmccandless** [comment]
this looks like it checks out. whats the plan to merge?

**Comment by kylecmccandless** [comment]
module_vite() should accept an array for the second parameter as well, right?

**Comment by razavistag** [comment]
This merge has not taken any effect as well as `Vite` configuration does not initiate on modules we newly created. @wikigods @kylecmccandless @wuwx @dydx @jnbn 
Please update the documentation. so developers can easily understand if there are any command to put for generate `vite` configuration

**NOT WORKING - SOLUTION DOES NOT TAKE EFFECT ON NEWLY CREATED PROJECT**

File: src/Commands/stubs/views/master.stub [hunk 1]
```diff
@@ -6,14 +6,14 @@
         <meta name="viewport" content="width=device-width, initial-scale=1">
         <title>Module $STUDLY_NAME$</title>
 
-       {{-- Laravel Mix - CSS File --}}
-       {{-- <link rel="stylesheet" href="{{ mix('css/$LOWER_NAME$.css') }}"> --}}
+       {{-- Laravel Vite - CSS File --}}
+       {{-- {{ module_vite('build-$LOWER_NAME$', 'Resources/assets/sass/app.scss') }} --}}
```
Comment (dcblogdev) [inline src/Commands/stubs/views/master.stub:10]: I don't think you need the {{ inside the {{ same for the closing ones

File: src/Commands/stubs/views/master.stub [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
 
     </head>
     <body>
         @yield('content')
 
-        {{-- Laravel Mix - JS File --}}
-        {{-- <script src="{{ mix('js/$LOWER_NAME$.js') }}"></script> --}}
+        {{-- Laravel Vite - JS File --}}
+        {{-- {{ module_vite('build-$LOWER_NAME$', 'Resources/assets/js/app.js') }} --}}
```
Comment (dcblogdev) [inline src/Commands/stubs/views/master.stub:17]: same as above

File: src/Commands/stubs/vite.stub [hunk 3]
```diff
@@ -0,0 +1,20 @@
+const dotenvExpand = require('dotenv-expand');
+dotenvExpand(require('dotenv').config({ path: '../../.env'/*, debug: true*/}));
+
+import { defineConfig } from 'vite';
+//import laravel from 'laravel-vite-plugin';
```
Comment (dcblogdev) [inline src/Commands/stubs/vite.stub:5]: the commented-out line, should it be active?

File: src/helpers.php [hunk 4]
```diff
@@ -34,3 +36,13 @@ function public_path($path = '')
         return app()->make('path.public') . ($path ? DIRECTORY_SEPARATOR . ltrim($path, DIRECTORY_SEPARATOR) : $path);
     }
 }
+
+if (! function_exists('module_vite')) {
+    /**
+     * support for vite
+     */
+    function module_vite($module, $asset): \Illuminate\Foundation\Vite
```
Comment (dcblogdev) [inline src/helpers.php:44]: vite has already been imported, you should be able to use `: Vite` here.

**Comment by wikigods** [comment]
I think it is ready

**Comment by anburocky3** [comment]
> I think it is ready

Can't wait to try those.

**Comment by MaticSulc** [comment]
Any plans to get this merged?

**Comment by ikramsyakir** [comment]
@wikigods can u show me demo how to use it? where should i run `npm run dev`?

**Pull request merged**